lib/mapObjects/CRewardableObject.cpp

```diff
@@ -161,9 +161,15 @@ void CRewardableObject::onHeroVisit(const CGHeroInstance *h) const
 						grantRewardWithMessage(rewards[CRandomGenerator::getDefault().nextInt(rewards.size()-1)]);
 						break;
 				}
-				return;
+				break;
 			}
 		}
+
+		if (getAvailableRewards(h).size() == 0)
```

Comment:
Uh... missed that this is going to cause crash for pickable objects.
Going to fix this soon.